llvm/test/CodeGen/X86/propogate-disjoint-in-shl-or.ll

```diff
@@ -0,0 +1,22 @@
+; NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+; RUN: llc -mtriple=x86_64 %s -start-before=x86-isel -o - -stop-after=x86-isel | FileCheck %s
+
+define void @add_shl_or_disjoint(i32 %x, ptr addrspace(1) %o) {
```

Comment:
I don't think this test should be using MIR. If there's an observable codegen change, this should just check the final ISA. A MIR test would be the right choice if it was only showing the flag survived through selection, but I don't see it here

Reply:
This commit actually aimed at propagating the disjoint flag in DAGCombine. I choose MIR test just want to check the result of the seletion, no any other concern or consideration. Maybe ll test is more suitable as your suggestion.